druid/src/widget/split.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A widget which splits an area in two, with a set ratio.
+
+use crate::kurbo::{Line, Point, Rect, Size};
+use crate::widget::flex::Axis;
+use crate::{
+    theme, BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx,
+    RenderContext, UpdateCtx, Widget, WidgetPod,
+};
+
+///An arbitrarily chosen constant. We may want to change this later, or make it a field.
+const SPLITTER_SIZE: f64 = 10.0;
+
+///A container containing two other widgets, splitting the area either horizontally or vertically.
+pub struct Split<T: Data> {
+    split_point: f64,
+    split_direction: Axis,
+    child1: WidgetPod<T, Box<dyn Widget<T>>>,
+    child2: WidgetPod<T, Box<dyn Widget<T>>>,
+}
+
+/// A builder for a widget which divides space horizontally
+/// The splitter bar will run top-to-bottom
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct HorizontalSplit;
```

Comment:
I'm not sure I love this pattern of having 'marker' widgets that delegate to another widget under the hood.

To me it might be easier to just have separate `Split::vertical()` and `Split::horizontal()` constructors on `Split` itself.

But this is mostly my preference; if other folks prefer it this way I'm not hugely bothered.

Reply:
I'm not sure how I liked the abstraction, but I copied the idea directly from `Flex`. I'm open to going with the `::vertical()` and `::horizontal()` constructors.

I'd like to pick a style and stick with it. I'm fairly serious about continuing to build layout widgets, and I don't want our style to be all over the place, if we can avoid it. 😊

Reply:
yep I agree consistency would be good, maybe something to discuss on zulip.